eodatasets/prepare/ncep_reanalysis_surface_pr_wtr.py

```diff
@@ -0,0 +1,166 @@
+"""
+Create index files from the command-line
+"""
+
+import datetime
+import uuid
+import urllib.parse
+from pathlib import Path
+
+import click
+import rasterio
+import rasterio.crs
+
+from eodatasets.serialise import write_yaml_from_dict
+from .utils import read_paths_from_file
+
+
+NOAA_WATER_VAPOUR_NS = uuid.UUID(hex='857bd048-8c86-4670-a2b4-5dbea26d7692')
+NOAA_DESCRIPTION = ''
+
+
+def get_coords(collection, xkey='lon', ykey='lat'):
+    """
+    transforms the bounds of a collection into an
+    encompassing bounding box. Keys for x and y are
+    customisable
+    """
+    return {
+        'll': {
+            ykey: collection.bounds.bottom,
+            xkey: collection.bounds.left
+        },
+        'lr': {
+            ykey: collection.bounds.bottom,
+            xkey: collection.bounds.right
+        },
+        'ul': {
+            ykey: collection.bounds.top,
+            xkey: collection.bounds.left
+        },
+        'ur': {
+            ykey: collection.bounds.top,
+            xkey: collection.bounds.right
+        }
+    }
+
+
+def get_uuid(collection, idx):
+    """
+    Returns a deterministic uuid based on band index and gdal checksum
+    """
+    origin = collection.tags()['NC_GLOBAL#References']
+
+    return uuid.uuid5(
+        NOAA_WATER_VAPOUR_NS,
+        '{}?{}'.format(origin, urllib.parse.urlencode({
+            'checksum': collection.checksum(idx),
+            'band_index': idx
+        }))
+    )
+
+
+def process_datasets(dataset: Path):
+    """
+    Generates a metadata document for each band available
+    in the water vapour file.
+
+    Each band is treated as a separate dataset since the
+    source file is updated in-place; allowing the file
+    to be modified but a static reference is provided for
+    each observation
+
+    """
+    datasets = []
+    creation_dt = (
+        datetime.datetime.utcnow()
+        .replace(tzinfo=datetime.timezone.utc)
+    )
+
+    with rasterio.open(str(dataset), 'r') as collection:
+        collection_start_date = (
+            datetime.datetime.strptime(
+                collection.tags()['time#units'],
+                'hours since %Y-%m-%d %H:%M:%S.%f'
+            )
+        )
+        creation_dt = (
```

Comment:
defined already a few lines above ([75](https://github.com/GeoscienceAustralia/eo-datasets/pull/42/files#diff-e4f346d610f818ad7bd62ee282ef5b18R75))

(pycharm is warning about this)